### PR TITLE
Guest login form fix

### PR DESCRIPTION
**Pull request by vrackovic**
- Fixed react-select to consist with project styles
- Form submission functionality implemented
- User authentication moved to `Panel` component so anonymous users can access applicant form
- Added TestGeneratorDiagram to gitignore

### CODE REVIEW

File: web/src/main/frontend/src/pages/GuestLogin.js [hunk 1]
```diff
@@ -10,45 +11,84 @@ class GuestLogin extends React.Component {
     this.props.category.fetchCategories();
   }
 
+  changeHandler = type => {
+    if (!type) {
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/GuestLogin.js]: Re-check if this is needed.

File: web/src/main/frontend/src/pages/Panel.js [hunk 2]
```diff
@@ -20,6 +27,19 @@ class Panel extends React.Component {
     {name: 'Login page', url: '/login', image: {name: LoginIcon}}
   ];
 
+  componentDidMount() {
+    this.props.auth.login().then(res => {
+      this.setState({
+        redirect:
+          res.status !== 200 ? (
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/Panel.js]: Invert this check: `res.status === 200 ? ...`

File: web/src/main/frontend/src/pages/Panel.js [hunk 3]
```diff
@@ -41,12 +61,12 @@ class Panel extends React.Component {
               </nav>
             </div>
             <Switch>
-              <Redirect exact from="/register" to="/dashboard" />
+              {this.state.redirect}
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/Panel.js]: Only save `status` in the state and then render the redirect based on that.

File: web/src/main/frontend/src/scss/components/_guestLogin.scss [hunk 4]
```diff
@@ -0,0 +1,104 @@
+.guest__form_fields {
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/components/_guestLogin.scss]: Please create an issue to clean up css in general.

File: web/src/main/frontend/src/pages/GuestLogin.js (hunk 1 above)
Comment (vrackovic) [reply web/src/main/frontend/src/pages/GuestLogin.js]: As it turns out, this check is needed. Even when `react-select` field is specified as `clearable=false`, and `X` doesn't exist, it is possible to clear it with pressing `Backspace`

File: web/src/main/frontend/src/pages/Panel.js (hunk 3 above)
Comment (vrackovic) [reply web/src/main/frontend/src/pages/Panel.js]: We found a better implementation of `Oauth`, so this part will probably be removed soon

**Comment by vrackovic** [comment]
Closing pull request until login is properly implemented

**Pull request closed without merging**